core/src/avm2/activation.rs

```diff
@@ -1033,6 +1034,24 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         Ok(FrameControl::Continue)
     }
 
+    fn op_in(&mut self) -> Result<FrameControl<'gc>, Error> {
+        let obj = self.context.avm2.pop().coerce_to_object(self)?;
+        let name = self.context.avm2.pop().coerce_to_string(self)?;
+
+        let mut has_prop = false;
+
+        if let Some(ns) = obj.resolve_any(name)? {
+            if !ns.is_private() {
+                let qname = QName::new(ns, name);
+                has_prop = obj.has_property(&qname)?;
+            }
```

Comment:
What about protected/internal/user-defined namespace? Seems like these return `false` in the Flash Player (https://github.com/adobe/avmplus/blob/master/core/Toplevel.cpp#L455)

Reply:
If I understand correctly, it makes sense to always qualify with public because AFAICT this opcode only concerns itself with public properties, even if the property might otherwise resolve in other open namespaces, .e.g:
```
package {
	public class Foo {
		protected var n:int;
		
		public function Foo() {
			trace("n" in this);
		}
	}
}

var foo:Foo = new Foo();
```
traces true with this PR but false in the official player.

Reply:
I did that to `in` for now. Should I also do this to `hasOwnProperty`, `propertyIsEnumerable`, or `setPropertyIsEnumerable`?